packages/core/src/tools/codegen/extract-components.ts

```diff
@@ -0,0 +1,77 @@
+import chalk from 'chalk';
+import path from 'path';
+import fs from 'fs';
+import { fetchBearerToken } from '../auth/fetch-bearer-token';
+import {
+  ExtractedFileType,
+  resolveComponentImportFiles,
+  sendCode,
+  validateConsent,
+  validateDeployContext,
+} from './utils';
+import { SitecoreConfig } from '../../config';
+
+export type ExtractComponentsConfig = {
+  scConfig: SitecoreConfig;
+  compilerOptions: {
+    [key: string]: unknown;
+  };
+  appFolder?: string;
+  componentMapPath?: string;
+};
+
+/**
+ * Extracts components from the app folder and sends them to XMCloud.
+ * @param {ExtractComponentsConfig} args - Arguments for extracting components
```

Comment:
It would be good to align the naming with our other build-time commands and methods for consistency.
```suggestion
 * @param {ExtractComponentsConfig} config - Config for components extraction
```